### PR TITLE
etcdserver: recover transport when recovering from a snapshot

### PR DESCRIPTION
**Pull request by xiang90**

### CODE REVIEW

**Comment by xiang90** [comment]
/cc @yichengq @Amit-PivotalLabs

fix #2244 #2287

File: etcdserver/server.go [hunk 1]
```diff
@@ -384,6 +384,13 @@ func (s *EtcdServer) run() {
 					log.Panicf("recovery store error: %v", err)
 				}
 				s.Cluster.Recover()
+
+				// recover raft transport
+				s.r.transport.RemoveAllPeers()
+				for _, m := range s.Cluster.Members() {
+					s.r.transport.AddPeer(m.ID, m.PeerURLs)
```
Comment (yichengq) [inline etcdserver/server.go]: it should not add itself into the transport..

Comment (xiang90) [inline etcdserver/server.go]: fixed

**Comment by yichengq** [comment]
lgtm

**Pull request merged**